concent_api/core/payments/bankster.py

```diff
@@ -264,80 +235,115 @@ def settle_overdue_acceptances(
     assert len(provider_ethereum_address) == ETHEREUM_ADDRESS_LENGTH
     assert provider_ethereum_address != requestor_ethereum_address
 
+    with transaction.atomic(using='control'):
+        requestor_client = Client.objects.get_or_create_full_clean(
+            public_key=requestor_public_key,
+        )
+        requestor_deposit_account = DepositAccount.objects.get_or_create_full_clean(
+            client=requestor_client,
+            ethereum_address=requestor_ethereum_address,
+        )
+
     # Bankster asks SCI about the amount of funds available in requestor's deposit.
     requestor_deposit_value = service.get_deposit_value(client_eth_address=requestor_ethereum_address)  # pylint: disable=no-value-for-parameter
 
-    # Concent defines time T0 equal to oldest payment_ts from passed SubtaskResultAccepted messages from
-    # subtask_results_accepted_list.
-    oldest_payments_ts = min(subtask_results_accepted.payment_ts for subtask_results_accepted in acceptances)
+    # Bankster begins a database transaction and puts a database lock on the DepositAccount object.
+    with transaction.atomic(using='control'):
+        DepositAccount.objects.select_for_update().get(
+            pk=requestor_deposit_account.pk
+        )
+
+        # Bankster sums the amounts of all existing DepositClaims that have the same payer as the one being processed.
+        sum_of_existing_requestor_claims = DepositClaim.objects.filter(
+            payer_deposit_account=requestor_deposit_account
+        ).aggregate(
+            sum_of_existing_claims=Coalesce(Sum('amount'), 0)
+        )
 
-    current_time = get_current_utc_timestamp()
+        # If the existing claims against requestor's deposit are greater or equal to his current deposit,
+        # we can't add a new claim.
+        if requestor_deposit_value <= sum_of_existing_requestor_claims['sum_of_existing_claims']:
+            return None
 
-    # Concent gets list of transactions from payment API where timestamp >= T0.
-    list_of_transactions = service.get_list_of_payments(  # pylint: disable=no-value-for-parameter
-        requestor_eth_address=requestor_ethereum_address,
-        provider_eth_address=provider_ethereum_address,
-        payment_ts=oldest_payments_ts,
-        current_time=current_time,
-        transaction_type=TransactionType.BATCH,
-    )
+        # Concent defines time T0 equal to oldest payment_ts from passed SubtaskResultAccepted messages from
+        # subtask_results_accepted_list.
+        oldest_payments_ts = min(subtask_results_accepted.payment_ts for subtask_results_accepted in acceptances)
```

Comment:
:+1: I like to see "senior code", if I did the same I would do it in coupe more lines and not so nice :)

Reply:
It is actually @kbeker code so good for him :)